src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -1110,6 +1118,288 @@ private async Task<HttpResponseMessage> GetDataSetMeta(
+                .DefaultDataSetVersion(filters: 1, indicators: 1, locations: 1, timePeriods: 3)
+                .WithStatus(dataSetVersionStatus)
+                .WithVersionNumber(major: 1, minor: 1, patch: 0)
+                .WithDataSet(dataSet)
+                .FinishWith(dsv => dataSet.LatestDraftVersion = dsv);
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.AddRange([latestDataSetVersion, requestedDataSetVersion]);
+                context.DataSets.Update(dataSet);
+            });
+
+            var response = await DownloadDataSet(dataSet.Id, requestedDataSetVersion.PublicVersion);
+
+            response.AssertForbidden();
+        }
+
+        [Fact]
+        public async Task DataSetDoesNotExist_Returns404()
+        {
+            var response = await DownloadDataSet(Guid.NewGuid());
+
+            response.AssertNotFound();
+        }
+
+        [Fact]
+        public async Task LatestVersionDoesNotExist_Returns404()
+        {
+            DataSet dataSet = DataFixture
+               .DefaultDataSet()
+               .WithStatusPublished();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            var response = await DownloadDataSet(dataSet.Id);
+
+            response.AssertNotFound();
+        }
+
+        [Fact]
+        public async Task RequestedVersionDoesNotExist_Returns404()
+        {
+            DataSet dataSet = DataFixture
+               .DefaultDataSet()
+               .WithStatusPublished();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion(filters: 1, indicators: 1, locations: 1, timePeriods: 3)
+                .WithStatusPublished()
+                .WithDataSet(dataSet)
+                .FinishWith(dsv => dataSet.LatestLiveVersion = dsv);
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.Add(dataSetVersion);
+                context.DataSets.Update(dataSet);
+            });
+
+            var response = await DownloadDataSet(dataSet.Id, "2.0");
+
+            response.AssertNotFound();
+        }
+
+        private async Task CreateGZippedTestCsv(DataSetVersion dataSetVersion, IReadOnlyList<TestClass> csvData)
+        {
+            var dataSetVersionPathResolver = TestApp.Services.GetRequiredService<IDataSetVersionPathResolver>();
+
+            using var memStream = new MemoryStream();
+            using var streamWriter = new StreamWriter(memStream);
+            using var csvWriter = new CsvWriter(streamWriter, CultureInfo.InvariantCulture);
+
+            csvWriter.WriteRecords(csvData);
+            await csvWriter.FlushAsync();
+            memStream.Position = 0;
```

Comment:
hmmmm, I'm sure I tested this and it wouldn't work without this line. But I've just tested it again and you're right, it can be removed. And I can also see the `SeekToBeginning` inside the `CompressToStream` method.

Removed :)